scaleoutsystems/stackn
# Bugfix for Issues 238

## Status


## Description

This hotfix is meant to address issue #238. Stackn CLI was rather unstable; so this hotfix aims at making the CLI more stable  and more user friendly via better feedback messages.

closes #238

## Types of changes


## Checklist

- [x] This pull request is against **develop** branch (not applicable for hotfixes)
- [x] I have included a link to the issue on GitHub or JIRA (if any)
- [x] I have read the [CONTRIBUTING](https://github.com/scaleoutsystems/stackn/blob/master/CONTRIBUTING.md) doc
- [x] I have added a reviewer for this pull request
- [x] I have added myself as an author for this pull request

## Further comments

Steps for reproducing the issue:

- Deploy a STACKn **from either main or the develop branch**
- Create at least one project template in order to be able to create a project and start apps
- Open a terminal either locally on your machine or within Jupyter in STACKn and try to run stackn cli commands such as: get, create and delete. Many should not work properly.

Steps for testing the patch code:

- Deploy a STACKn **from either main or the develop branch**
- Create at least one project template in order to be able to create a project and start apps
- Open a terminal either, again locally or in a jupyter lab in STACKn, and clone this issue branch, because you need to install the patched cli version.
- For testing use, it's better to create a python virtual env . Activate the new stackn-cli-test virtual env and install stackn cli. So:

```bash
$ python3 -m venv stackn-cli-test
$ source stackn-cli-test/bin/activate
(stackn-cli-test) $ cd stackn/cli
(stackn-cli-test) $ pip install .
```
**NOTE**: if you get dependecies error, it can be related to the fact that you started a new clean virtual environments. So install needed dependencies via pip if need be.

- Check if you have a json config file under the path `~/.scaleout/stack.json`. If so, delete it so that you can test what happens if you are not logged in via stackn cli

- Test create commands
- Test get commands
- Test delete an app (above all, which was related to issue #238)

**Note**: this is not by far a complete version of the CLI. The aim was rather to make it more stable and fix the bug with deleting apps.
